### PR TITLE
EG-969, synced PageBuilder updates from viewer, EG now uses pagesWithValidation()

### PR DESCRIPTION
**Pull request by cmrhmrc**
Synced changes from the Viewer

### CODE REVIEW

File: it/endpoints/GetApprovalProcessISpec.scala [hunk 1]
```diff
@@ -83,16 +83,16 @@ class GetApprovalProcessISpec extends IntegrationSpec {
 
   "Calling the approval GET process code endpoint with a valid ID" should {
 
-    def populateDatabase(processToSave: JsValue): String = {
-      lazy val request = buildRequest("/external-guidance/approval/2i-review")
+    // def populateDatabase(processToSave: JsValue): String = {
```
Comment (LesWilson) [inline it/endpoints/GetApprovalProcessISpec.scala:86]: Can this code be removed?

File: test/services/PageBuilderSpec.scala [hunk 2]
```diff
@@ -618,14 +621,148 @@ class PageBuilderSpec extends BaseSpec with ProcessJson with StanzaHelper {
...
+
+          testSqpFap(indexedSeqOfPages(2))
 
-          testSqpSap(indexedSeqOfPages(2))
 
         case Left(err) => //fail(s"Flow error $err")
       }
     }
 
+    // "Return ungrouped Instruction stanzas when text contents do not start with similar text" in {
```
Comment (LesWilson) [inline test/services/PageBuilderSpec.scala:633]: Can we remove this commented out code?

File: it/endpoints/GetApprovalProcessISpec.scala (hunk 1 above)
Comment (cmrhmrc) [reply it/endpoints/GetApprovalProcessISpec.scala:86]: done

File: test/services/PageBuilderSpec.scala (hunk 2 above)
Comment (cmrhmrc) [reply test/services/PageBuilderSpec.scala:633]: These are bullet point tests which should be moved out of the PageBuilderSpec class in the viewer code, but until this is done, its easier to do the comparison between the viewer code and backend code when syncing. I've been leaving the move to Richard as he fully understands the bullet point stuff

**Review by LesWilson — APPROVED** [review]

**Pull request merged**